src/lib/operators/abstract_read_only_operator.hpp

```diff
@@ -31,6 +31,9 @@ class AbstractReadOnlyOperator : public AbstractOperator {
     virtual ~AbstractReadOnlyOperatorImpl() = default;
     virtual std::shared_ptr<const Table> _on_execute() = 0;
   };
+
+  // for access to _on_execute
+  friend class AbstractOperator;
```

Comment:
Could you explain why this is needed, please?

Reply:
The `_on_execute` method without a transaction context only exists in this class, so `AbstractOperator` would not know about it.

Reply:
ping @mjendruk